app/javascript/components/auth-key-pair-cloud/auth-key-pair-cloud-form.schema.js

```diff
@@ -0,0 +1,48 @@
+import { componentTypes, validatorTypes } from '@data-driven-forms/react-form-renderer';
+import { API } from '../../http_api';
+
+const providerOptions = () => API.get(`/auth_key_pair_cloud/ems_form_choices`).then(data => [
+  ...data.ems_choices.map(({ id, name }) => ({ value: id, label: name })),
+
+]);
+
+function createSchema() {
+  const fields = [
+    {
+      id: 'name',
+      name: 'name',
+      label: __('Name'),
+      component: componentTypes.TEXT_FIELD,
+      validateOnMount: true,
+      validate: [{
+        type: validatorTypes.REQUIRED,
+      }, {
+        type: validatorTypes.PATTERN,
+        message: __('Please provide a valid git URL'),
+      }],
+    }, {
+      id: 'public_key',
+      name: 'public_key',
+      label: __('Public Key (optional)'),
```

Comment:
Let's stick to the current convention for now, set the `isRequired` attribute to `true` for required fields and don't mark the optional ones.